src/main/java/org/fogbowcloud/manager/core/OrderStateTransitioner.java

```diff
@@ -85,7 +90,13 @@ public static void deactivateOrder(Order order) throws UnexpectedException {
                         "Tried to remove order %s from the active orders but it was not active", order.getId());
                 throw new UnexpectedException(message);
             }
+
             closedOrders.removeItem(order);
+
+            // TODO: is it here? (only used for closed orders, here the resource is definitely deleted on provider)
+            order.setInstanceId(null);
```

Comment:
@llvieira I did't quite understand why not adding the setInstance(null) in the CloseProcessor instead of here (there is at least one advantage of doing there: we release cloud resources in the CloseProcessor)

Reply:
The method `deactivateOrder(Order order)` where that piece of code is located is just called by the `ClosedProcessor`. Therefore, it already works fine!